integration/nested-repeated-values-optional-all/nested-repeated-values-optional-all.ts

```diff
@@ -0,0 +1,165 @@
+// Code generated by protoc-gen-ts_proto. DO NOT EDIT.
+// source: nested-repeated-values-optional-all.proto
+
+/* eslint-disable */
+import { BinaryReader, BinaryWriter } from "@bufbuild/protobuf/wire";
+import { StringValue } from "./google/protobuf/wrappers";
+
+export const protobufPackage = "";
+
+export interface NestedList {
+  aString?: string[] | undefined;
+}
+
+export interface Example {
+  list?: NestedList | undefined;
+}
+
+function createBaseNestedList(): NestedList {
+  return { aString: [] };
+}
+
+export const NestedList: MessageFns<NestedList> = {
+  encode(message: NestedList, writer: BinaryWriter = new BinaryWriter()): BinaryWriter {
+    if (message.aString !== undefined && message.aString.length !== 0) {
+      for (const v of message.aString) {
+        StringValue.encode({ value: v!! }, writer.uint32(10).fork()).join();
+      }
+    }
+    return writer;
+  },
+
+  decode(input: BinaryReader | Uint8Array, length?: number): NestedList {
+    const reader = input instanceof BinaryReader ? input : new BinaryReader(input);
+    let end = length === undefined ? reader.len : reader.pos + length;
+    const message = createBaseNestedList();
+    while (reader.pos < end) {
+      const tag = reader.uint32();
+      switch (tag >>> 3) {
+        case 1:
+          if (tag !== 10) {
+            break;
+          }
+
+          const el = StringValue.decode(reader, reader.uint32()).value;
+          if (el) {
```

Comment:
@vecerek if `el` was an empty string, we probably want to still `.push` it? I.e. should we just for undefined instead of truthiness?

Reply:
Yes, you're correct! I'll make that change and I've just noticed I didn't push the re-generated TS files and I noticed another problem related to multiple `case` statements:

```
Cannot redeclare block-scoped variable 'el'.ts(2451)
```

I'll have to wrap the cases in a block to address this issue.

Reply:
@stephenh I've just force-pushed the rest of the changes. Thank you for the swift review!